### PR TITLE
Fix function quote warning

### LINKED ISSUE #15443: Warning regarding `helm-spacemacs-help.el` lambda
The messages buffer contains the following warning.

```{emacs-lisp}
.spacemacs.d/layers/+completion/helm/local/helm-spacemacs-help/helm-spacemacs-help.el: Warning: (lambda nil \...) quoted with ' rather than with #' [2 times]
```

### PR DESCRIPTION
**Pull request by bryce-carson**
Change `'` to `#'`, resolves #15443

### CODE REVIEW

**Comment by lebensterben** [comment]
I don't think quoting is needed.

**Comment by bryce-carson** [comment]
> I don't think quoting is needed.

Perhaps not, this was just a quick fix of the warning.

**Review by lebensterben — CHANGES_REQUESTED** [review]
#' is unnecessary

**Review by smile13241324 — APPROVED** [review]

**Comment by bryce-carson** [comment]
I made these new commits and tested the fix on this branch, based on the develop branch. "Works on my machine."

I can now install a candidate from the Helm Spacemacs Layer sources with S-return. :)

It took a bit to figure out that the actions are actually zero-indexed. 🙄

**Comment by bryce-carson** [comment]
I followed the rebase instructions in the CircleCI check for my branch. Everything should be good now, unless the instructions misled me or I introduced a problem into the PR.

File: layers/+completion/helm/local/helm-spacemacs-help/helm-spacemacs-help.el [hunk 1]
```diff
@@ -192,12 +192,12 @@
...
-                                          ;; Add Layer
-                                          (helm-select-nth-action 3)))
-    (define-key map (kbd "<M-return>") '(lambda () (interactive)
-                                          ;; Open packages.el
-                                          (helm-select-nth-action 1)))
+    (define-key map (kbd "<S-return>") (lambda ()
+                                           "Install a layer, the current Helm candidate."
+                                           (interactive) (helm-select-nth-action 5)))
+    (define-key map (kbd "<M-return>") (lambda ()
+                                           "Open the `packages.el' file of a layer, the current Helm candidate."
```
Comment (lebensterben) [inline layers/+completion/helm/local/helm-spacemacs-help/helm-spacemacs-help.el:199]: indentation doesn't seem correct.
press = = to reindent it.

File: layers/+completion/helm/local/helm-spacemacs-help/helm-spacemacs-help.el [hunk 2]
```diff
@@ -192,12 +192,12 @@
...
     (set-keymap-parent map helm-map)
-    (define-key map (kbd "<S-return>") '(lambda () (interactive)
-                                          ;; Add Layer
-                                          (helm-select-nth-action 3)))
-    (define-key map (kbd "<M-return>") '(lambda () (interactive)
-                                          ;; Open packages.el
-                                          (helm-select-nth-action 1)))
+    (define-key map (kbd "<S-return>") (lambda ()
+                                         "Install a layer, the current Helm candidate."
+                                         (interactive) (helm-select-nth-action 5)))
```
Comment (lebensterben) [inline layers/+completion/helm/local/helm-spacemacs-help/helm-spacemacs-help.el:197]: why is it changed from 3 to 5?

Comment (bryce-carson) [reply layers/+completion/helm/local/helm-spacemacs-help/helm-spacemacs-help.el:197]: The actions are zero indexed. 3 was wrong, given the comment that did exist which was to "add layer", meaning install it.

**Review by smile13241324 — APPROVED** [review]
Looks good I will merge this in the next view days but first I want to review the other linked lambda quote and merge them both.

**Pull request merged**